src/include/duckdb/common/serializer/binary_common.hpp

```diff
@@ -0,0 +1,23 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/common/serializer/binary_common.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#include "duckdb/common/typedefs.hpp"
+
+#pragma once
+
+namespace duckdb {
+
+enum class BinaryMessageKind : uint8_t {
```

Comment:
Can we call this `BinaryMessageLength` or `BinaryMessageType` instead of `Kind`? And is `Message` what we call `Field`? Maybe even `BinaryFieldLenght` or `BinaryFieldType`? I also see you using `frame` for a field in some places.

Reply:
I agree naming can be more consistent. I generally prefer `Kind` as a discriminator as `Type` is a crazily overloaded term, [also what LLVM standards suggest](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly), although I can change it to BinaryMessageType to be more in line with DuckDB.